cli/tests/extensions.rs

```diff
@@ -280,6 +280,135 @@ fn conflicting_extension_name_is_filtered() {
     assert!(output.contains("extension was filtered out"));
 }
 
+mod permissions {
+    use std::ffi::OsStr;
+
+    use assert_cmd::assert::Assert;
+    use predicates::prelude::*;
+
+    use super::*;
+
+    struct TestCLI {
+        tempdir: TempDir,
+        cwd: Option<PathBuf>,
+    }
+
+    impl Default for TestCLI {
+        fn default() -> Self {
+            Self { tempdir: TempDir::new().unwrap(), cwd: None }
+        }
+    }
+
+    impl TestCLI {
+        fn new() -> Self {
+            Default::default()
+        }
+
+        fn cwd(mut self, cwd: PathBuf) -> Self {
+            self.cwd = Some(cwd);
+            self
+        }
+
+        fn install_extension(&self, path: &Path) -> Assert {
+            self.run(&["extension", "install", "-y", &path.to_string_lossy()])
+        }
+
+        fn run<S: AsRef<str> + AsRef<OsStr>>(&self, args: &[S]) -> Assert {
+            let mut cmd = Command::cargo_bin("phylum").unwrap();
+
+            cmd.env("XDG_DATA_HOME", self.tempdir.path()).args(args);
+
+            if let Some(cwd) = self.cwd.as_ref() {
+                cmd.current_dir(cwd);
+            }
+
+            cmd.assert()
+        }
+    }
+
+    #[test]
+    fn permission_dialog_is_shown_without_yes_flag() {
+        let test_cli = TestCLI::new().cwd(fixtures_path().join("permissions"));
+
+        test_cli
+            .run(&[
+                "extension",
+                "install",
+                &fixtures_path().join("permissions").join("correct-read-perms").to_string_lossy(),
+            ])
+            .failure()
+            .stdout(predicate::str::contains("Read from the following paths"))
+            .stderr(predicate::str::contains("Can't ask for permissions"));
+    }
+
+    #[test]
+    fn correct_read_permission_successful_install_and_run() {
+        let test_cli = TestCLI::new().cwd(fixtures_path().join("permissions"));
+
+        test_cli
+            .install_extension(&fixtures_path().join("permissions").join("correct-read-perms"))
+            .success();
+
+        test_cli
+            .run(&["correct-read-perms"])
+            .success()
+            .stdout(predicate::str::contains("await Deno.readFile"));
```

Comment:
I don't understand, why should this output `await Deno.readFile`?

Reply:
Because `main.ts` is reading from `main.ts` and printing it out. 😅

Reply:
Oh wow... That just confused me, got it.